shaman/util.py

```diff
@@ -28,20 +28,26 @@ def get_next_node():
     return None
 
 
-def is_node_healthy(node):
+def is_node_healthy(node, only_check=False):
```

Comment:
this looks like it could now be two functions. Something like `check_node_health` and `node_healthiness` (or some other name?): one checks only, and the other ones does the bookkeeping. `healthiness` can use/call `check_node_health` 

I never thought we would need these separated, but now that there is a use case we should split them up